rasterio/_base.pyx

```diff
@@ -233,31 +245,30 @@ cdef class DatasetBase(object):
                     k = k.lstrip("+")
                     crs[k] = v
 
-            _gdal.CPLFree(proj_c)
+            _gdal.CPLFree(proj)
             _gdal.OSRDestroySpatialReference(osr)
         else:
             log.debug("GDAL dataset has no projection.")
         return crs
 
     def read_transform(self):
+        """Return the stored GDAL GeoTransform"""
```

Comment:
Since this will be in the public API as a Dataset method, we should change the method name to indicate that it is a *GDAL* transform. Maybe also consider making it private, `_read_gdal_transform`?